### PR TITLE
Split apart objects.rs

### PR DESCRIPTION
**Pull request by ltratt**
`objects.rs` was becoming a monster (1000 lines) and is only going to grow. This PR minimally splits things apart (https://github.com/softdevteam/yksom/commit/16009256db15a18199d1dea44fca217cd2a31e97) and then makes a few minor fix-ups.

I'm not totally sure if this is the best way of splitting it up, so I'm open to suggestions!

### CODE REVIEW

File: src/lib/vm/objects/integers.rs [hunk 1]
```diff
@@ -7,6 +7,10 @@
 // at your option. This file may not be copied, modified, or distributed except according to those
 // terms.
 
+//! yksom has three ways of representing integers: as tagged integer (inside a `Val`); boxed
```
Comment (ptersilie) [inline src/lib/vm/objects/integers.rs]: Why are tagged integers (`Val`) not inside this module as well?

Comment (ltratt) [reply src/lib/vm/objects/integers.rs]: `Val` is the union of tagged integers and boxed objects, so it's a very separate thing (in its own file `val.rs`). In a sense, `Val` stands outside the normal object system, if that makes sense.

Comment (ptersilie) [reply src/lib/vm/objects/integers.rs]: It does. Thanks for the clarification.

**Comment by ptersilie** [comment]
Do you want to squash anything or is this ready for merging as is?

**Comment by ltratt** [comment]
Merge as-is please!

**Pull request merged**